File: android/app/src/main/kotlin/com/agent/aios/domain/agent/AuditLog.kt
package com.agent.aios.domain.agent

import com.agent.aios.domain.model.ToolAuditEntry

class AuditLog(private val maxSize: Int = 100) {
    private val entries = mutableListOf<ToolAuditEntry>()

    fun add(
        tool: String,
        args: String,
        risk: com.agent.aios.domain.model.ToolRisk,
        approved: Boolean,
        result: String,
    ) {
        synchronized(entries) {
            entries.add(ToolAuditEntry(System.currentTimeMillis(), tool, args, risk, approved, result))
            while (entries.size > maxSize) {
                entries.removeAt(0)
            }
        }
    }

    fun getAll(): List<ToolAuditEntry> = synchronized(entries) { entries.toList() }

    fun clear() = synchronized(entries) { entries.clear() }
}

File: lib/domain/agent/audit_log.dart
import 'package:aios/domain/entities/agent_models.dart';

class AuditLog {
  AuditLog({int maxSize = 100}) : _maxSize = maxSize;

  final int _maxSize;
  final List<ToolAuditEntry> _entries = [];

  void add(
    String tool,
    String args,
    ToolRisk risk, {
    required bool approved,
    required String result,
  }) {
    _entries.add(
      ToolAuditEntry(
        timestamp: DateTime.now().millisecondsSinceEpoch,
        tool: tool,
        args: args,
        risk: risk,
        approved: approved,
        result: result,
      ),
    );
    while (_entries.length > _maxSize) {
      _entries.removeAt(0);
    }
  }

  List<ToolAuditEntry> getAll() => List.unmodifiable(_entries);

  void clear() => _entries.clear();
}
